src/opencl/image_function_opencl.cpp

```diff
@@ -1144,6 +1164,62 @@ namespace Image_Function_OpenCL
         multiCL::launchKernel2D( kernel, width, height );
     }
 
+    void SetPixel( Image & image, uint32_t x, uint32_t y, uint8_t value )
+    {
+        Image_Function::ParameterValidation( image );
+
+        const multiCL::OpenCLProgram & program = GetProgram();
+        multiCL::OpenCLKernel kernel( program, "setPixelOpenCL" );
+
+        if ( image.empty() || x >= image.width() || y >= image.height() )
+            throw imageException( "Bad input parameters in image function" );
+
+        const uint32_t width = image.width();
+        const uint32_t height = image.height();
+        const uint32_t rowSize = image.rowSize();
+        const uint32_t offset = x * rowSize + y;
+
+        kernel.setArgument( image.data(), offset, rowSize, width, height, value );
+
+        multiCL::launchKernel2D( kernel, image.width(), image.height() );
```

Comment:
Why not just launch a single thread instead?

Reply:
Not quite sure, but based on the `opencl_helper` the SetPixel should be run with 2 dimensional kernel, since it has the (X, Y) points. I based it on how many coordinate the function accepts. If it had a `z` coordinate, I would choose 3D. Please correct me if I am doing it wrong.

Reply:
Sure. But it is only 1 point you are writing. Given an x and a y, you can calculate the 1D position by `y * width + x` . It's probably better to not occupy more resources than needed. Although, i guess it hardly makes a difference.

Reply:
Sure! I will take a look at it again, maybe you have some idea on how to do the SetPixel with Vector Points?

Reply:
Just call it as `multiCL::launchKernel1D( kernel, 1 );`